fund.go

```diff
@@ -0,0 +1,219 @@
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	tempPath, _ := url.Parse(path.Join(request.Id, "transactions?command=deposit"))
+	path := client.HostName.ResolveReference(tempPath).String()
+	req, err := http.NewRequest("POST", path, bytes.NewBuffer(b))
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("fineract-platform-tenantid", "default")
+	// TODO: Add a better way to generate this authorisation
+	req.Header.Set("Authorization", "Basic bWlmb3M6cGFzc3dvcmQ=")
+
+	var resp *http.Response
+	errTry := highbrow.Try(5, func() error {
+		resp, err = client.HttpClient.Do(req)
+		return err
+	})
+	if errTry != nil {
+		rawMessage := json.RawMessage([]byte(errTry.Error()))
+		return nil, &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		rawMessage := json.RawMessage(body)
+		return nil, &FineractError{GetFineractStatusCode(resp.StatusCode), &rawMessage}
+	}
+
+	var response FundIncrementResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		rawMessage := json.RawMessage([]byte(err.Error()))
+		return nil, &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	return &response, err
+}
+
+func (client *Client) FundDecrement(request FundDecrementRequest) (*FundDecrementResponse, error) {
+	b, err := json.Marshal(request)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	tempPath, _ := url.Parse(path.Join(request.Id, "transactions?command=withdrawal"))
+	path := client.HostName.ResolveReference(tempPath).String()
+	req, err := http.NewRequest("POST", path, bytes.NewBuffer(b))
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("fineract-platform-tenantid", "default")
+	// TODO: Add a better way to generate this authorisation
+	req.Header.Set("Authorization", "Basic bWlmb3M6cGFzc3dvcmQ=")
+
+	var resp *http.Response
+	errTry := highbrow.Try(5, func() error {
+		resp, err = client.HttpClient.Do(req)
+		return err
+	})
+	if errTry != nil {
+		rawMessage := json.RawMessage([]byte(errTry.Error()))
+		return nil, &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		rawMessage := json.RawMessage(body)
+		return nil, &FineractError{GetFineractStatusCode(resp.StatusCode), &rawMessage}
+	}
+
+	var response FundDecrementResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		rawMessage := json.RawMessage([]byte(err.Error()))
+		return nil, &FineractError{ErrCodeSerialization, &rawMessage}
+	}
+
+	return &response, err
+}
+
+func (client *Client) GetFundValue(request FundValueRequest) (*FundValueResponse, error) {
+	tempPath, _ := url.Parse(request.Id)
+	req, _ := http.NewRequest("GET", client.HostName.ResolveReference(tempPath).String(), nil)
```

Comment:
This is nearly identical code for all the methods. Please move this to a common method on *Client.